libpf/pfelf/file.go

```diff
@@ -49,8 +50,12 @@ const (
 	// parsed sections (e.g. symbol tables and string tables; libxul
 	// has about 4MB .dynstr)
 	maxBytesLargeSection = 16 * 1024 * 1024
+	buildIDSectionName   = ".note.gnu.build-id"
```

Comment:
how do we treat go build id, which might be in a differently named note? I think it's maybe `.note.go.build-id` or something?

Reply:
This is handled in: https://github.com/DataDog/otel-profiling-agent/blob/967d1d580e39d8f63ca4dbf18b60f6755287a47a/libpf/pfelf/file.go#L485-L499 

I had to extract the string for the GNU build id section to a constant since the new `HasDwarfData()` function uses it, and the linter was yelling at me for string constants duplication 🤦

Reply:
minor: the absence of build id being an error is slightly too high in log level

Reply:
👍  Right now the log error is only emitted if the ELF has dwarf data but no build id, so it should be pretty rare. I'm tempted to leave it as is, and happy to follow-up on this in an ulterior PR